### PR TITLE
#4685 [Manager] Unable to delete and rename default workspace name in quick create workspace

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4685)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/CreateWorkspace/SetupWorkspaceForm.js [hunk 1]
```diff
@@ -27,17 +27,28 @@ import * as validate from 'utils/validate';
...
   const formValues = hooks.useReduxFormValues(FORM.CREATE_WORKSPACE);
   const workspaces = hooks.useAllWorkspaces();
 
   React.useEffect(() => {
-    if (!formValues?.workspace?.name)
+    if (!formValues?.workspace?.name && !anyTouched) {
       change(FORM.CREATE_WORKSPACE, 'workspace.name', defaultWorkspaceName);
-  }, [change, defaultWorkspaceName, formValues]);
+      // Reset `anyTouched` so we won't run it again when users are editing the form
+      touch(FORM.CREATE_WORKSPACE);
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/CreateWorkspace/SetupWorkspaceForm.js:49]: @eechih I've tried various ways to fix this. This is by far the easiest and cleanest way for me and it works. Let me know if there's a better approach of fixing this :)

File: ohara-manager/client/src/hooks/createWorkspaceHooks.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const useCreateWorkspaceState = () => {
 export const useUniqueWorkspaceName = (prefix = 'workspace') => {
   const workspaces = hooks.useAllWorkspaces();
   return useMemo(() => {
-    for (var postfix = 1; postfix <= Number.MAX_SAFE_INTEGER; postfix++) {
+    for (let postfix = 1; postfix <= Number.MAX_SAFE_INTEGER; postfix++) {
```
Comment (konekoya) [inline ohara-manager/client/src/hooks/createWorkspaceHooks.js:46]: var is evil...

Comment (eechih) [reply ohara-manager/client/src/hooks/createWorkspaceHooks.js:46]: I think so.

File: ohara-manager/client/src/components/Workspace/CreateWorkspace/SetupWorkspaceForm.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/CreateWorkspace/SetupWorkspaceForm.js:49]: @konekoya 辛苦了～

**Comment by konekoya** [comment]
Thanks for the review @eechih 😄

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Pull request merged**